SwiftFlow/CoreTypes/MainStore.swift

```diff
@@ -38,16 +38,20 @@ public class MainStore: Store {
     public required init(reducer: AnyReducer, appState: StateType, middleware: [Middleware]) {
         self.reducer = reducer
         self.appState = appState
-        self.dispatchFunction = self._defaultDispatch
 
         // Wrap the dispatch function with all middlewares
-        self.dispatchFunction = middleware.reverse().reduce(self.dispatchFunction) {
+        self.dispatchFunction = middleware.reverse().reduce(self._defaultDispatch) {
             dispatchFunction, middleware in
                 return middleware(self.dispatch, { self.appState })(dispatchFunction)
         }
     }
 
     public func subscribe(subscriber: AnyStoreSubscriber) {
+        guard subscribers.indexOf({ $0 === subscriber }) == nil else {
+            print("Store subscriber is already added, ignoring.")
+            return;
+        }   
+        
```

Comment:
Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)